src/server/pachyderm_test.go

```diff
@@ -86,6 +86,80 @@ func testJob(t *testing.T, shards int) {
 	}
 }
 
+func TestDuplicatedJob(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration tests in short mode")
+	}
+
+	t.Parallel()
+
+	ctx := context.Background()
+	pachClient := getPachClient(t)
+
+	dataRepo := uniqueString("TestDuplicatedJob.data")
+	require.NoError(t, pfsclient.CreateRepo(pachClient, dataRepo))
+
+	commit, err := pfsclient.StartCommit(pachClient, dataRepo, "", "")
+	require.NoError(t, err)
+
+	fileContent := "foo\n"
+	_, err = pfsclient.PutFile(pachClient, dataRepo, commit.ID, "file", 0, strings.NewReader(fileContent))
+	require.NoError(t, err)
+
+	require.NoError(t, pfsclient.FinishCommit(pachClient, dataRepo, commit.ID))
+
+	pipelineName := uniqueString("TestDuplicatedJob.pipeline")
+	cmd := []string{"cp", path.Join("/pfs", dataRepo, "file"), "/pfs/out/file"}
+	require.NoError(t, ppsclient.CreatePipeline(
+		pachClient,
+		pipelineName,
+		"",
+		cmd,
+		nil,
+		1,
+		[]*ppsclient.PipelineInput{{Repo: &pfsclient.Repo{Name: dataRepo}}},
+	))
+
+	// Wait for the pipeline to pick up the job
+	time.Sleep(10 * time.Second)
```

Comment:
In general I like to avoid waiting for state with sleep unless it's really hard to get around.
I think this test would be just as good if you just manually created 2 identical jobs and then checked to make sure that it only resulted in 1 job actually being created.
We have other tests that check that pipelines create jobs so it's fine to tighten the scope of this test.